### ISSUE TITLE
for var = expr1 to expr2 do expr3 done

### ISSUE THREAD

**Issue by kevmal** [dotnet/fsharp#6586]
Seems identifiers between `var` and `=` simply get ignored. For example the following compiles/runs:
```fsharp 
for i bleh bleh this runs fine = 1 to 20 do 
    printfn "%d" i
```

**Comment by cartermp** [comment]
Even more fun:

```fsharp
let foo x = x

for i foo 12 = 1 to 20 do 
    printfn "%d" i
```

Prints 1-12 in FSI on a new line, just as your repro.

Definitely feels like a bug 😄

**Comment by dsyme** [comment]
How odd.  Anyway fix is at https://github.com/Microsoft/visualfsharp/pull/6621

**Comment by cartermp** [comment]
Re-opening as this is not resolved in VS

**Comment by cartermp** [comment]
Confirmed as fixed in 16.2.
